h/static/scripts/directive/test/annotation-test.js

```diff
@@ -355,6 +355,202 @@ describe('annotation', function() {
     });
   });
 
+  describe('link', function () {
+    var link = require('../annotation').link;
+
+    /** Return Angular's $rootScope. */
+    function getRootScope() {
+      var $rootScope;
+      inject(function(_$rootScope_) {
+        $rootScope = _$rootScope_;
+      });
+      return $rootScope;
+    }
+
+    function newScope() {
+      return getRootScope().$new();
+    }
+
+    function mockElement() {
+      return {
+        on: sinon.stub()
+      };
+    }
+
+    function mockAttributes() {
+      return;  // Left undefined because link() doesn't use it.
+    }
+
+    function mockAnnotationController() {
+      return {
+        editing: sinon.stub().returns(false),
+        onKeydown: "annotationController.onKeydown"  // Sentinel value.
+      };
+    }
+
+    function mockThreadController() {
+      return {
+        collapsed: true,
+        toggleCollapsed: sinon.stub()
+      };
+    }
+
+    function mockThreadFilterController() {
+      return {
+        active: sinon.stub(),
+        freeze: sinon.stub()
+      };
+    }
+
+    function mockDeepCountController() {
+      return {
+        count: sinon.stub()
+      };
+    }
+
+    it('binds AnnotationController.onKeydown to "keydown"', function () {
+      var annotationController = mockAnnotationController();
```

Comment:
There is an awful lot of boilerplate at the start of each of these tests. Can you try to extract it out and more DRY?